client/blocks/editor-checkout-modal/index.tsx

```diff
@@ -82,29 +82,31 @@ const EditorCheckoutModal = ( props: Props ) => {
 		: cartData.products.map( ( product ) => product.product_slug );
 	const commaSeparatedProductSlugs = productSlugs?.join( ',' ) || null;
 
-	return hasEmptyCart ? null : (
-		<Modal
-			open={ isOpen }
-			overlayClassName="editor-checkout-modal"
-			onRequestClose={ onClose }
-			title=""
-			shouldCloseOnClickOutside={ false }
-			icon={ <Icon icon={ wordpress } size={ 36 } /> }
-		>
-			<ShoppingCartProvider cartKey={ cartKey } getCart={ wpcomGetCart } setCart={ wpcomSetCart }>
-				<StripeHookProvider
-					fetchStripeConfiguration={ fetchStripeConfigurationWpcom }
-					locale={ props.locale }
-				>
-					<CompositeCheckout
-						isInEditor
-						siteId={ site.ID }
-						siteSlug={ site.slug }
-						productAliasFromUrl={ commaSeparatedProductSlugs }
-					/>
-				</StripeHookProvider>
-			</ShoppingCartProvider>
-		</Modal>
+	return (
+		isOpen && (
+			<Modal
+				open={ isOpen }
+				overlayClassName="editor-checkout-modal"
+				onRequestClose={ onClose }
+				title=""
```

Comment:
I know it was already like this before this PR, but should provide a title?

Reply:
Are you thinking at a11y? In this case, yes, a title would be good. But not sure if we should display it or not on screen. Cc: @ollierozdarz

Reply:
A way of improving the a11y without changing the current design could be:
1. Add a modal title
2. Hide it by using the`onboarding-screen-reader-text()` SCSS mixin from the [@automattic/onboarding package](https://github.com/Automattic/wp-calypso/blob/trunk/packages/onboarding/styles/mixins.scss#L239)

This is similar to the approach that we took for the [Focused Launch modal](https://github.com/Automattic/wp-calypso/blob/fd38eb47d0effef6a796e6548625b33dd7177d58/packages%2Flaunch%2Fsrc%2Flaunch%2Fstyles.scss#L60-L62)

Reply:
It's not straight forward to re-use the onboarding mixins since they bring also[ this z-index function](https://github.com/Automattic/wp-calypso/blob/trunk/packages/onboarding/styles/base-styles.scss#L1) (takes 1 argument) which is in conflict with [the one used in this file](https://github.com/Automattic/wp-calypso/blob/trunk/client/blocks/editor-checkout-modal/style.scss#L6) (two arguments). So an ideal solution would be to refactor onboarding package first to get out that import and explicitly add it only where needed.

Reply:
Hi @ciampo, 

> Are you thinking at a11y? In this case, yes, a title would be good.
> But not sure if we should also display it or not on screen. Cc: @ollierozdarz 

What is the title here referring to?

Reply:
This would be the modal title, when displaying Checkout at the end of the launch flow. But at the moment we hide it (so it's there just for screen readers) and we display only the W icon as originally designed.

<img width="1157" alt="Screenshot 2020-11-26 at 09 14 45" src="https://user-images.githubusercontent.com/14192054/100319421-3c813f00-2fc8-11eb-907f-96df54a8e4b5.png">

Reply:
To clarify further:
- at the beginning of this conversation the modal title was simply left empty
- @razvanpapadopol, then, made changes so that the modal title reads "Checkout modal" — although it's only accessible by screen readers, and is not visible in the browser

We wanted to confirm with you if this is a good approach, or if you think that we should instead also visually show the title in the browser

Reply:
> It's not straight forward to re-use the onboarding mixins since they bring also this z-index function (takes 1 argument) which is in conflict with the one used in this file (two arguments). So an ideal solution would involve refactoring onboarding package first to get out that import and explicitly add it only where needed.

Also, I opened https://github.com/Automattic/wp-calypso/issues/47805 to keep track of this task

Reply:
Thanks @razvanpapadopol and @ciampo. I like the change to add the title for screen readers! (although I'd lose the word 'modal' personally). Great catch. 

I'd like to keep it to have no title for non-screen readers so the page is ultra focused on the task at hand (selecting and converting). When there are more 'things', there is more noise.

Reply:
Thank you @ollierozdarz ! I opened #47829 to track this task separately from this PR. This conversation can be marked as resolved